src/Drivers/SoftDeleteDriver.php

```diff
@@ -80,14 +80,17 @@ public function persist(Model $model): bool
          *
          * @see https://github.com/cloudcreativity/laravel-json-api/issues/371
          */
-        if ($this->willSoftDelete($model)) {
-            $column = $model->getDeletedAtColumn();
-            // save the original date so we can put it back later on.
-            $deletedAt = $model->{$column};
-            // delete the record so that deleting and deleted events get fired.
-            $model->delete();
-            // apply the original date back before saving, so that we keep date provided by the client.
-            $model->{$column} = $deletedAt;
+         if ($this->willSoftDelete($model)) {
+             $column = $model->getDeletedAtColumn();
+             // save the original date so we can put it back later on.
+             $deletedAt = $model->{$column};
+             // delete the record so that deleting and deleted events get fired.
+             $response = $model->delete();  // capture the response 
+             // if everything worked out then update the $deletedAt
+             if ($response !== false){   
```

Comment:
So the `persist()` method that this code is in is type-hinted as returning a boolean. It's meant to return `false` if anything goes wrong with the persisting changes.

Here, if `$response === false`, we should return `false`. That's because the persistence has not happened, which from the JSON:API package perspective is an error. That's because at this point in the code, we know the client has asked for the resource to be soft-deleted, but the model has not been soft deleted. Therefore, we are in an error state as the server has not done what the client has asked.

So I believe the code should be:

```php
if ($response === false) {
    return false;
}
```

Reply:
Thank you for looking over the PR.  I just committed the change.